controller/search_user_blackbox_test.go

```diff
@@ -0,0 +1,227 @@
+	s.controller = NewSearchController(s.svc, s.db, s.Configuration)
+}
+
+func (s *TestSearchUserSearch) SetupTest() {
+	s.clean = cleaner.DeleteCreatedEntities(s.DB)
+}
+
+func (s *TestSearchUserSearch) TearDownTest() {
+	s.clean = cleaner.DeleteCreatedEntities(s.DB)
+}
+
+type userSearchTestArgs struct {
+	pageOffset *string
+	pageLimit  *int
+	q          string
+}
+
+type userSearchTestExpect func(*testing.T, okScenarioUserSearchTest, *app.SearchResponseUsers)
+type userSearchTestExpects []userSearchTestExpect
+
+type okScenarioUserSearchTest struct {
+	name                  string
+	userSearchTestArgs    userSearchTestArgs
+	userSearchTestExpects userSearchTestExpects
+}
+
+func (s *TestSearchUserSearch) TestUsersSearchOK() {
+
+	idents := s.createTestData()
+	defer s.cleanTestData(idents)
+
+	tests := []okScenarioUserSearchTest{
+		{"With uppercase fullname query", userSearchTestArgs{s.offset("0"), limit(10), "TEST_AB"}, userSearchTestExpects{s.totalCount(1)}},
+		{"With uppercase fullname query", userSearchTestArgs{s.offset("0"), limit(10), "TEST_AB"}, userSearchTestExpects{s.totalCount(1)}},
+		{"With uppercase email query", userSearchTestArgs{s.offset("0"), limit(10), "EMAIL_TEST_AB"}, userSearchTestExpects{s.totalCount(1)}},
+		{"With lowercase email query", userSearchTestArgs{s.offset("0"), limit(10), "email_test_ab"}, userSearchTestExpects{s.totalCount(1)}},
+		{"with special chars", userSearchTestArgs{s.offset("0"), limit(10), "&:\n!#%?*"}, userSearchTestExpects{s.totalCount(0)}},
+		{"with * to list all", userSearchTestArgs{s.offset("0"), limit(10), "*"}, userSearchTestExpects{s.totalCountAtLeast(len(idents))}},
+		{"with multi page", userSearchTestArgs{s.offset("0"), limit(10), "TEST"}, userSearchTestExpects{s.hasLinks("Next")}},
+		{"with last page", userSearchTestArgs{s.offset(strconv.Itoa(len(idents) - 1)), limit(10), "TEST"}, userSearchTestExpects{s.hasNoLinks("Next"), s.hasLinks("Prev")}},
+		{"with different values", userSearchTestArgs{s.offset("0"), s.limit(10), "TEST"}, userSearchTestExpects{s.differentValues()}},
+	}
+
+	for _, tt := range tests {
+		_, result := test.UsersSearchOK(s.T(), context.Background(), s.svc, s.controller, tt.userSearchTestArgs.pageLimit, tt.userSearchTestArgs.pageOffset, tt.userSearchTestArgs.q)
+		for _, userSearchTestExpect := range tt.userSearchTestExpects {
+			userSearchTestExpect(s.T(), tt, result)
+		}
+	}
+}
+
+func (s *TestSearchUserSearch) TestUsersSearchBadRequest() {
+	t := s.T()
+	tests := []struct {
+		name               string
+		userSearchTestArgs userSearchTestArgs
+	}{
+		{"with empty query", userSearchTestArgs{s.offset("0"), limit(10), ""}},
+	}
+
+	for _, tt := range tests {
+		test.UsersSearchBadRequest(t, context.Background(), s.svc, s.controller, tt.userSearchTestArgs.pageLimit, tt.userSearchTestArgs.pageOffset, tt.userSearchTestArgs.q)
+	}
+}
+
+func (s *TestSearchUserSearch) createTestData() []account.Identity {
+	names := []string{"TEST_A", "TEST_AB", "TEST_B", "TEST_C"}
```

Comment:
Let's test different combinations: Specific username. Specific email. Specific Full Name. Combination of specific name which is a part of the full name and the email.